beets/dbcore/types.py

```diff
@@ -45,26 +64,26 @@ class Type:
     """
 
     @property
-    def null(self) -> model_type:
+    def null(self) -> N:
         """The value to be exposed when the underlying value is None.
         """
-        return self.model_type()
+        return cast(N, self.model_type())
```

Comment:
It's overall very interesting that we ended up with null values that don't match the "main" type… like, I know it's not the case, but you would imagine that the null value would always agree with the underlying Python type that the `Type` represents. Maybe this is a good nudge to revisit this in the future and someday make this the case, which could reduce some unintentional weirdness.

On the other hand, maybe it is exactly what we want, in the sense that the actual Python type of the underlying values should be considered to be `Union[N, T]`. That lets us deal with `T` in the common case but occasionally provide `N`, knowing that `Union[None, T]` is the same as `Optional[T]`, and `Union[T, T]` is the same as `T`.

Reply:
I think currently, we can't avoid this: We do have types where e.g. `T = int` and `N = None`, i.e.

> in the sense that the actual Python type of the underlying values should be considered to be `Union[N, T]`

Maybe there's a world where we would have a more generic way for values to be either of `model_type` or `None`, instead of having the various explicit NULL-able and on-NULL-able types? Actually, I don't really know my way around beets auto-tagging super well and can't tell at the moment why we can't just allow all types to be `None` (as in not set).

Reply:
The `cast` here is a particularly ugly hack: In the abstract `Type` class, `N` and `T` may in fact be incompatible. This default implementation only works for the case `N = T`. More cleanly, this should be done in a subclass. Alternatively, we could add a `null_type` similar to `model_type` and use a `Protocol` to restrict it's constructor signature.

For now, I think I'll leave it as-is to keep the code simple.